### DISCUSSION TITLE
conditional face/vertex selection not working

### DISCUSSION THREAD

**Discussion by rhit-jiaoq** [cnr-isti-vclab/PyMeshLab#315] [Q&A]
ms.apply_filter('compute_selection_by_condition_per_face', condselect=cond)
pymeshlab.pmeshlab.PyMeshLabException: Failed to apply filter: compute_selection_by_condition_per_face
Filter does not exists. Take a look at MeshSet.print_filter_list function.

Is it removed?

**Comment by alemuntoni** [comment]
Nope, the filter still exists. I don't know why it is not available in your installation.

**Reply by rhit-jiaoq** [reply]
I installed via the .whl file under actions tab. It seems like only the newest two versions are available, but the problem exists in both. Some .dll are missing when I import pymeshlab. The program compiles normally, but could it possibly cause the problem? I am pasting the error info here

>>> import pymeshlab
Warning:
Unable to load the following plugins:

        filter_func.dll: filter_func.dll does not seem to be a Qt Plugin.

Cannot load library C:\Users\jiaoq\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\pymeshlab\lib\plugins\filter_func.dll: The specified module could not be found.
        filter_mesh_booleans.dll: filter_mesh_booleans.dll does not seem to be a Qt Plugin.

Cannot load library C:\Users\jiaoq\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\pymeshlab\lib\plugins\filter_mesh_booleans.dll: The specified module could not be found.
        filter_sketchfab.dll: filter_sketchfab.dll does not seem to be a Qt Plugin.

Cannot load library C:\Users\jiaoq\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\pymeshlab\lib\plugins\filter_sketchfab.dll: The specified module could not be found.
        io_3ds.dll: io_3ds.dll does not seem to be a Qt Plugin.

Cannot load library C:\Users\jiaoq\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\pymeshlab\lib\plugins\io_3ds.dll: The specified module could not be found.
        io_e57.dll: io_e57.dll does not seem to be a Qt Plugin.

Cannot load library C:\Users\jiaoq\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\pymeshlab\lib\plugins\io_e57.dll: The specified module could not be found.
        io_u3d.dll: io_u3d.dll does not seem to be a Qt Plugin.

Cannot load library C:\Users\jiaoq\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\pymeshlab\lib\plugins\io_u3d.dll: The specified module could not be found.

Can you double check or maybe send me some older version of pymeshlab?

**Reply by alemuntoni** [reply]
Which whl are you downloading? The one from the `main` or from the `devel` branch?
Anyway, I think something is going wrong in your system, in our tests all the plugins are correctly loaded in a clean environment. 
Could you please try to install the package from pip and see if the same errors appear?

**Reply by rhit-jiaoq** [reply]
I installed via pip but the same problem occurs. I managed to download the 2021.10 release for 3.9 and that one has conditional face selection. Guess I will use that one. Thank you.

**Comment by jmespadero** [comment]
I have just tested this. Everything seems ok starting from a fresh linux machine.

```bash
python3 -m venv pm
source pm/bin/activate
pip3 install pymeshlab
```
Output from installation
```
Collecting pymeshlab
  Using cached pymeshlab-2022.2.post4-cp310-cp310-manylinux_2_31_x86_64.whl (64.9 MB)
Collecting numpy
  Downloading numpy-1.25.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (17.6 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 17.6/17.6 MB 1.9 MB/s eta 0:00:00
Installing collected packages: numpy, pymeshlab
Successfully installed numpy-1.25.1 pymeshlab-2022.2.post4
```

Command to test `compute_selection_by_condition_per_face` filter (Both errors are expected)
```
>>> import pymeshlab as ml
>>> ms = ml.MeshSet()
>>> ms.apply_filter('compute_selection_by_condition_per_face', condselect="1")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
pymeshlab.pmeshlab.PyMeshLabException: compute_selection_by_condition_per_face requires at least one mesh loaded in the MeshSet.
>>> ms.compute_selection_by_condition_per_face(condselect="1")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/jespa/proyectos/pm/lib/python3.10/site-packages/pymeshlab/__init__.py", line 41, in foo
    res_dict = self.apply_filter(name, **kwargs)
pymeshlab.pmeshlab.PyMeshLabException: compute_selection_by_condition_per_face requires at least one mesh loaded in the MeshSet.

```
